stdlib/3/fcntl.pyi

```diff
@@ -76,20 +81,39 @@ LOCK_SH: int
 LOCK_UN: int
 LOCK_WRITE: int
 
-# TODO All these return either int or bytes depending on the value of
-# cmd (not on the type of arg).
-def fcntl(fd: FileDescriptorLike,
-          cmd: int,
-          arg: Union[int, bytes] = ...) -> Any: ...
+@overload
+def fcntl(__fd: FileDescriptorLike,
+          __cmd: int,
+          __arg: int = ...) -> int: ...
+@overload
+def fcntl(__fd: FileDescriptorLike,
+          __cmd: int,
+          __arg: bytes) -> bytes: ...
 # TODO This function accepts any object supporting a buffer interface,
 # as arg, is there a better way to express this than bytes?
-def ioctl(fd: FileDescriptorLike,
-          request: int,
-          arg: Union[int, bytes] = ...,
-          mutate_flag: bool = ...) -> Any: ...
-def flock(fd: FileDescriptorLike, operation: int) -> None: ...
-def lockf(fd: FileDescriptorLike,
-          cmd: int,
-          len: int = ...,
-          start: int = ...,
-          whence: int = ...) -> Any: ...
+@overload
+def ioctl(__fd: FileDescriptorLike,
+          __request: int,
+          __arg: int = ...,
+          __mutate_flag: bool = ...) -> int: ...
+@overload
+def ioctl(__fd: FileDescriptorLike,
+          __request: int,
+          __arg: bytearray,
+          __mutate_flag: Literal[True] = ...) -> int: ...
+@overload
```

Comment:
I think this overload can be omitted (note that `bytearray` is treated as a subtype of `bytes` to the typechecker).